daml-script/dump/src/main/scala/com/daml/script/dump/TreeUtils.scala

```diff
@@ -151,6 +154,49 @@ object TreeUtils {
   def createdReferencedCids(ev: CreatedEvent): Set[ContractId] =
     ev.createArguments.foldMap(args => args.fields.foldMap(f => valueCids(f.getValue.sum)))
 
+  /** A simple event causes the creation of a single contract and returns its contract id.
+    *
+    * A create event is a simple event. An exercise event can be a simple event.
+    */
+  case class SimpleEvent(event: TreeEvent.Kind, contractId: ContractId)
```

Comment:
This doesn’t take into account whether the contract ids are used right? So if an exercise creates two contracts and returns one and we use only that, we still will don’t hit this?

That seems very sensible, just making sure I understand the logic correctly.

Reply:
Yes, that's right. It's a relatively simple heuristic.